src/main/java/core/basesyntax/model/Activity.java

```diff
@@ -0,0 +1,63 @@
+package core.basesyntax.model;
+
+import java.util.Arrays;
+
+public class Activity {
+    private static final int ACTIVITY_TYPE_INDEX = 0;
+    private static final int FRUIT_NAME_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+    private static final String SEPARATOR = ",";
+
+    private final Type activityType;
+    private final String fruitName;
+    private final Integer quantity;
+
+    public Activity(Type activityType, String fruitName, Integer quantity) {
+        this.activityType = activityType;
+        this.fruitName = fruitName;
+        this.quantity = quantity;
+    }
+
+    public static Activity makeActivity(String activityLine) {
```

Comment:
I suppose it's some type of factory method, but in this case it's definitely not the best choice to use this pattern. What if tomorrow you would need support other file formats and protocols, then your model class should know about all of them and have more factory methods, it's a break of SRP, so extract this logic from here.